plugin.rb

```diff
@@ -551,6 +564,19 @@ def event_ics(opts = {})
 
       render body: cal.to_ical, formats: [:ics], content_type: Mime::Type.lookup("text/calendar") unless performed?
     end
+
+    # Logging in with user API keys normally only works by passing certain headers.
+    # As we cannot force third-party software to send those headers, we need to fake
+    # them using request parameters.
+    def current_user
```

Comment:
@haveyaseen Happy to be proven wrong, but in my testing appending &api_username=\<username\>&api_key=\<user_api_key\> works fine with user api credentials. 

For example:

Url (feel free to use this to test):
```
webcal://thepavilion.io/c/test/l/calendar.ics?time_zone=Australia/Melbourne&api_username=angus45&api_key=075ce4f7f1d68f83d91d542f9aa2513840e919f5b4b9c3c3746b552b96bc3bc3
```
Result:

<img width="574" alt="Screenshot at Sep 04 10-01-26" src="https://user-images.githubusercontent.com/5931623/64216227-33a3d980-cefb-11e9-876b-aab2e73eb35c.png">
<img width="108" alt="Screenshot at Sep 04 10-01-00" src="https://user-images.githubusercontent.com/5931623/64216223-33a3d980-cefb-11e9-8dd3-24101cdfba36.png">

Moreoever, if you look at [lib/auth/default_current_user_provider.rb](https://github.com/discourse/discourse/blob/master/lib/auth/default_current_user_provider.rb) where the api key and username are checked the headers and parameters seem to be used as alternates.

Reply:
@angusmcleod It would seem to me that that is not a `UserApiKey` (even if it may appear so under `/admin/api/keys`). In fact even `ApiKey`s are assigned a user which is why you. It may surprise you that it's not even necessary to provide the `api_username` for an `ApiKey` to be used. For example the following URL can be used to fetch your private messages:
```
https://thepavilion.io/topics/private-messages/angus45.json?api_key=075ce4f7f1d68f83d91d542f9aa2513840e919f5b4b9c3c3746b552b96bc3bc3
```
I would advise you to delete that key as it exposes other nasty details as well. Won't post those here but you get the idea I think.

Reply:
@haveyaseen Yup, I'm aware what it exposes. It's a dummy user account, so I'm not concerned ;) But yes! I prefer the approach you're suggesting, so all good as far as I'm concerned. Thanks for taking the time to explain what you're thinking :+1: